libs/api/metadata-converter/src/lib/iso19115-3/read-parts.ts

```diff
@@ -1,6 +1,197 @@
-import { pipe } from '../function-utils'
-import { extractCharacterString } from '../iso19139/read-parts'
+import {
+  findChildElement,
+  findChildrenElement,
+  findNestedElement,
+  findNestedElements,
+  readAttribute,
+  XmlElement,
+} from '../xml-utils'
+import {
+  ChainableFunction,
+  combine,
+  flattenArray,
+  getAtIndex,
+  map,
+  mapArray,
+  pipe,
+} from '../function-utils'
+import {
+  extractCharacterString,
+  extractRole,
+  extractUrl,
+} from '../iso19139/read-parts'
+import {
+  Individual,
+  Organization,
+  RecordKind,
+} from '@geonetwork-ui/common/domain/model/record'
+
+export function readKind(rootEl: XmlElement): RecordKind {
+  return pipe(
+    findNestedElement(
+      'mdb:metadataScope',
+      'mdb:MD_MetadataScope',
+      'mdb:resourceScope',
+      'mcc:MD_ScopeCode'
+    ),
+    readAttribute('codeListValue'),
+    map(
+      (scopeCode): RecordKind =>
+        scopeCode === 'service' ? 'service' : 'dataset'
+    )
+  )(rootEl)
+}
+
+export function findIdentification() {
+  return (rootEl: XmlElement) => {
+    const kind = readKind(rootEl)
+    let eltName = 'mri:MD_DataIdentification'
+    if (kind === 'service') eltName = 'srv:SV_ServiceIdentification'
+    return findNestedElement('gmd:identificationInfo', eltName)(rootEl)
+  }
+}
+
+export function findDistribution() {
+  return findNestedElement('mdb:distributionInfo', 'mrd:MD_Distribution')
+}
+
+// from cit:CI_Responsibility
+export function extractOrganization(): ChainableFunction<
+  XmlElement,
+  Organization
+> {
+  const getOrganization = findNestedElement('cit:party', 'cit:CI_Organisation')
+  const getUrl = pipe(
+    getOrganization,
+    findNestedElements(
+      'cit:contactInfo',
+      'cit:CI_Contact',
+      'cit:onlineResource',
+      'cit:CI_OnlineResource',
+      'cit:linkage'
+    ),
+    getAtIndex(0),
+    extractUrl()
+  )
+  return pipe(
+    combine(
+      pipe(
+        getOrganization,
+        findChildElement('cit:name', false),
+        extractCharacterString()
+      ),
+      getUrl
+    ),
+    map(([name, website]) => ({
+      name,
+      ...(website && { website }),
+    }))
+  )
+}
+
+// from cit:CI_Responsibility
+export function extractIndividuals(): ChainableFunction<
+  XmlElement,
+  Array<Individual>
+> {
+  const getRole = pipe(findChildElement('cit:role'), extractRole())
+  const getIndividual = pipe(
+    findChildElement('cit:individual'),
+    findChildElement('cit:CI_Individual')
+  )
+  const getPosition = pipe(
+    getIndividual,
+    findChildElement('cit:positionName'),
+    extractCharacterString()
+  )
+  const getNameParts = pipe(
+    getIndividual,
+    findChildElement('cit:name'),
+    extractCharacterString(),
+    map((fullName) => {
+      if (!fullName) return []
+      const parts = fullName.split(/\s+/)
+      if (!parts.length) return [fullName, null]
+      const first = parts.shift()
+      return [first, parts.join(' ')]
```

Comment:
Out of curiosity, and for further reference, is it the standard to split only on the first space?

Reply:
Right, good point... I was also wondering about that myself. It's not standard at all and depending on cultures it can give wrong results. On the other hand this is not critical, as the firstName and lastName fields are just there for convenience and are concatenated back when writing (actually I even wonder if it's a good idea to have those and not just a "fullName" field...)